src/main/java/net/imglib2/algorithm/fill/ComparableComparator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * #%L
+ * ImgLib2: a general-purpose, multidimensional image processing library.
+ * %%
+ * Copyright (c) 2009 - 2016, Tobias Pietzsch, Stephan Preibisch, Stephan Saalfeld,
+ * John Bogovic, Albert Cardona, Barry DeZonia, Christian Dietz, Jan Funke,
+ * Aivar Grislis, Jonathan Hale, Philipp Hanslovsky, Grant Harris, Stefan Helfrich,
+ * Mark Hiner, Martin Horn, Steffen Jaensch, Lee Kamentsky, Larry Lindsey,
+ * Melissa Linkert, Mark Longair, Brian Northan, Nick Perry, Dimiter Prodanov,
+ * Curtis Rueden, Johannes Schindelin, Jean-Yves Tinevez and Michael Zinsmaier.
+ * All rights reserved.       
+ *                            
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *                            
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *                            
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *                            
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.imglib2.algorithm.fill;
+
+import net.imglib2.util.Pair;
+
+import java.util.Comparator;
+
+/**
+ * @author Philipp Hanslovsky &lt;hanslovskyp@janelia.hhmi.org&gt;
+ *
+ *
+ * Default implementation of {@link Comparable} for the use in {@link FloodFill} when source and target are both
+ * {@link net.imglib2.RandomAccessible<Comparable>}
+ *
+ */
+public class ComparableComparator< T extends Comparable< T >, U extends Comparable< U >  >
```

Comment:
I think we also want an `EqualsComparator` that works with `Object.equals`.  That would be the most generic thing possible with a simple Interface.  The bad news is that neither of the ImgLib2 `Types` overrides `equals` in a sensible way, and so all the generalization effort leads nowhere.  `Comparable<T>` implies an order in the value domain of `T`, i.e. we cannot use it for `ARGBType`.  I am currently working on a branch of ImgLib2 that will implement `Type.valueEquals` as an alternative to `Object.equals` which is used in other place s with different assumptions.